enhancements/windows-containers/bring-your-own-windows-host.md

```diff
@@ -0,0 +1,196 @@
+  needs to be an authorized key in the instance.
+* WMCO will then perform all the required steps within the VM for it to be
+  added to the cluster as an OpenShift worker node and removed from it.
+* [Handle upgrades](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
+  of all components installed by WMCO.
+
+### Non-Goals
+
+* Installing the container runtime in the Windows instance
+* Managing the Windows operating system(OS) including handling of OS updates
+* OpenShift Builds
+* OpenShift DeploymentConfigs
+
+## Proposal
+
+The cluster admin will create a ConfigMap called _windows-instances_ in the
+_openshift-windows-machine-config-operator_ namespace. WMCO will be configured
+to watch for this special ConfigMap, whose data section will have the following
+layout:
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: windows-instances
+  namespace: openshift-windows-machine-config-operator
+data:
+  10.1.42.1: |-
+    username=Administrator
+  instance.dns.com: |-
+    username=core
+```
+WMCO will configure or de-configure instances based on the contents of the data
+section and will use the [cloud-private-key]((https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator.md#design-details)
+to communicate with them. The key value above can either be the IP address of
+the instance or it's DNS name.
+
+### User Stories
+
+#### Story 1
+As an OpenShift cluster administrator I would like to add an existing Windows
+instance to the cluster as a worker node.
+
+#### Story 2
+As an OpenShift cluster administrator I would like to remove an existing Windows
+instance from the cluster as a worker node.
+
+#### Story 3
+As an OpenShift cluster administrator I would like to have the Kubernetes
+components on Windows nodes upgraded when new Kubernetes versions or patch
+versions are released.
+
+### Risks and Mitigations
+
+The same risk and mitigation that apply to Windows nodes added using MachineSets
+to an IPI cluster applies here. In addition there is potential for a node to
+become unstable if configuration fails and will require administrator
+intervention to fix. There is no real mitigation here other that for us to test
+this feature thoroughly in CI.
+
+From an implementation stand point there is potential for code complexity given
+we are attempting to track the services being used in the code itself. While
+this might not be an issue today given the number of services, it could get
+magnified as we add to the number of services. The mitigation for this would
+be to externally track the services using a Kubernetes object. This however is
+beyond the scope of this enhancement and should be tackled holistically across
+MachineSet initiated and BYOH nodes in a separate enhancement.
```

Comment:
Theoretically all these services could be run as `DaemonSets` however the answer here is **none** because all Windows container images are packaged with a Windows kernel and Red Hat has a policy to not ship 3rd party kernels for support reasons. Plus we would need our build systems to support building Windows images.